op-batcher/batcher/driver.go

```diff
@@ -292,46 +302,59 @@ func (l *BatchSubmitter) loop() {
 	for {
 		select {
 		case <-ticker.C:
-			l.loadBlocksIntoState(l.ctx)
-
-		blockLoop:
-			for {
-				l1tip, err := l.l1Tip(l.ctx)
-				if err != nil {
-					l.log.Error("Failed to query L1 tip", "error", err)
-					break
-				}
-				l.recordL1Tip(l1tip)
-
-				// Collect next transaction data
-				txdata, err := l.state.TxData(l1tip.ID())
-				if err == io.EOF {
-					l.log.Trace("no transaction data available")
-					break // local for loop
-				} else if err != nil {
-					l.log.Error("unable to get tx data", "err", err)
-					break
-				}
-				// Record TX Status
-				if receipt, err := l.txMgr.SendTransaction(l.ctx, txdata.Bytes()); err != nil {
-					l.recordFailedTx(txdata.ID(), err)
-				} else {
-					l.recordConfirmedTx(txdata.ID(), receipt)
-				}
-
-				// hack to exit this loop. Proper fix is to do request another send tx or parallel tx sending
-				// from the channel manager rather than sending the channel in a loop. This stalls b/c if the
-				// context is cancelled while sending, it will never fully clear the pending txns.
-				select {
-				case <-l.ctx.Done():
-					break blockLoop
-				default:
-				}
+			// prioritize the `done` condition over the ticker, even though select ordering is randomized
+			select {
+			case <-l.done:
```

Comment:
@sebastianst  I don't think we need this. Even if `l.done` would be hit here, we would hit it when we enter `l.publishStateToL1`. We should also cancel the `l.loadCtx` early on in the shutdown process.

Reply:
Given that `l.cancelLoad()` is called immediately prior to `close(l.done)`, we can be sure that `l.loadBlocksIntoState(l.loadCtx)` will run with a cancelled context whenever `l.done` is readable. We can probably safely revert this code back to the previous simpler state without the explicit `done` prioritization